src/wic.py

```diff
@@ -0,0 +1,71 @@
+ of the data'''
+
+def read_wic_data():
+    # PDF has 97 pages. Skip page 0 because it shows Statewide totals which we don't need
+    parse_wic_pdf("data_folder/illinois_wic_data_january_2021.pdf", "final_jsons/wic.csv", 1, 96)
+
+
+def parse_wic_pdf(source_pdf_filepath: str, destination_csv_filepath: str, first_page_zero_indexed: int, last_page_zero_indexed: int):
+    # use regular expression matching to find lines that start with certain words and save as variables:
+    # find rows that start with Total (this includes Total Women, Total Infant and Total Children rows)
+    Total_re = re.compile("Total")
+    # It's not clear specifically what "LA Total" means, but these rows contains the subtotal values for the specific County
+    County_Total_re = re.compile("LA Total")
+    # find rows that start with three digits (these rows contain County ID and name, example: 031 COOK)
+    County_re = re.compile(r"\d\d\d")
+
+    column_names = ["County_ID", "County", "WIC1", "WIC2", "Amer. Indian or Alaskan Native",
+                    "Asian", "Black or African American", "Native Hawaii or Other Pacific Isl.",
+                    "White", "Multi-Racial", "Total Participants", "Hispanic or Latino"]
+
+    # create empty data frame with column names
+    data = pd.DataFrame(columns=column_names)
+
+    with pdfplumber.open(source_pdf_filepath) as pdf:
+    
+        # range() excludes the last value, so add 1 to include the last page
+        for page_num in range(first_page_zero_indexed, last_page_zero_indexed + 1):
+            page = pdf.pages[page_num]
+
+            # extract_text() adds spaces where the horizontal distance between bits of text is greater than x_tolerance
+            # and adds newline characters where the vertical distance between bits of text is greater than y_tolerance.
+            text = page.extract_text(x_tolerance=2, y_tolerance=0)
+
+            # iterate thru each line on a page
+            for line in text.split("\n"):
+                if County_re.match(line):
+                    # We have to find the County information first because we insert it in every row
+                    # maxsplit=1 because some counties have spaces in their name, example: Jo Daviess
+                    County = (line.split(sep=" ", maxsplit=1)) 
+                elif Total_re.match(line):
+                    new_line = (line.split(sep=" ")) # This looks something like ["Total", "Women", 1, 2, 3, 4]
+                    new_line = County + new_line
+                    data = data.append(pd.Series(new_line, index=data.columns), ignore_index=True)
+                elif County_Total_re.match(line):
+                    new_line = (line.split(sep=" ")) # This looks something like ["LA", "Total", 1, 2, 3, 4]
+                    new_line = County + new_line
+                    data = data.append(pd.Series(new_line, index=data.columns), ignore_index=True)
```

Comment:
I guess it works, but I wouldn't overwrite the variable here like this. For a list, append returns None, so it is a bit confusing what happens here.

Looks like pd.concat might be a better function for this.